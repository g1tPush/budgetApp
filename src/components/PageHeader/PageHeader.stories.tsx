import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import PageHeader from './PageHeader'

export default {
    title: 'Page Header',
    component: PageHeader
} as ComponentMeta<typeof PageHeader>

const Template: ComponentStory<typeof PageHeader> = (args) => <PageHeader {...args} />

export const Header = Template.bind({})

Header.args = {
    headerType: 'dashboard'
}
