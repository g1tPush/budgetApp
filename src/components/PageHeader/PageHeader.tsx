import React from 'react'
import Button from '../Button/Button'
import TotalSign from '../TotalSign/TotalSign'
import { Link } from 'react-router-dom'

interface PageHeaderProsp {
    /**
     * The header type
     */
    headerType: 'dashboard' | 'create' | 'edit',
    numOfExpenses?: number,
    totalSum?: number
}

const PageHeader: React.FC<PageHeaderProsp> = ({ headerType, numOfExpenses, totalSum }) => {
    if (headerType === 'dashboard') {
        return (
            <>
                <TotalSign numOfExpenses={numOfExpenses} totalSum={totalSum} />
                <Link to='/create' data-testid='addExpense'>
                    <Button label='Add Expense' style={{ marginTop: '20px' }} />
                </Link>
            </>
        )
    } else if (headerType === 'create') {
        return (
            <TotalSign title={'Create Expense'} />
        )
    } else {
        return (
            <TotalSign title={'Edit Expense'} />
        )
    }
}

export default PageHeader