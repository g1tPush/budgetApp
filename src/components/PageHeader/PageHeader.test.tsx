import React from 'react'
import renderer from 'react-test-renderer'
import PageHeader from './PageHeader'
import { BrowserRouter } from 'react-router-dom'

it('renders dashboard', () => {
    const tree = renderer
        .create(
            <BrowserRouter>
                <PageHeader headerType='dashboard' numOfExpenses={1} totalSum={1000} />
            </BrowserRouter>
        )
        .toJSON()
    expect(tree).toMatchSnapshot()
})

it('renders create sign', () => {
    const tree = renderer
        .create(<PageHeader headerType='create' />)
        .toJSON()
    expect(tree).toMatchSnapshot()
})

it('renders edit sign', () => {
    const tree = renderer
        .create(<PageHeader headerType='edit' />)
        .toJSON()
    expect(tree).toMatchSnapshot()
})