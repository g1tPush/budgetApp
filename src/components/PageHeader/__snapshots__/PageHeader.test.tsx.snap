// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders create sign 1`] = `
<p
  className="sc-gsnTZi gAzqOS"
>
  Create Expense
</p>
`;

exports[`renders dashboard 1`] = `
Array [
  <p
    className="sc-gsnTZi gAzqOS"
  >
    Viewing 
    <b>
      1
    </b>
     expense totalling 
    <b>
      $
      1000
    </b>
  </p>,
  <a
    data-testid="addExpense"
    href="/create"
    onClick={[Function]}
  >
    <button
      className="sc-bczRLJ dhHktu"
      mode={0}
      style={
        Object {
          "marginTop": "20px",
        }
      }
    >
      Add Expense
    </button>
  </a>,
]
`;

exports[`renders edit sign 1`] = `
<p
  className="sc-gsnTZi gAzqOS"
>
  Edit Expense
</p>
`;
